db/collab.go

```diff
@@ -68,6 +78,42 @@ func (s *Session) Broadcast(msg Message) error {
 	return mostRecentError
 }
 
+func (s *Session) BroadcastError(uid string, err string) error {
+	errorMsg := Message{
+		Author: uid,
+		Type:   "ERROR",
+		Body:   err,
+	}
+	if broadcastErr := s.BroadcastTo(errorMsg, uid); broadcastErr != nil {
+		return broadcastErr
+	}
+	return nil
+}
+
+func (s *Session) BroadcastTo(msg Message, uids ...string) error {
+	var mostRecentError error
+	s.Lock.Lock()
+	defer s.Lock.Unlock()
+	for _, uid := range uids {
+		if err := websocket.JSON.Send(s.Conns[uid], msg); err != nil {
+			mostRecentError = err
+		}
+	}
+	return mostRecentError
+}
```

Comment:
This is more semantically clear:

```suggestion
func (s *Session) BroadcastTo(msg Message, uids ...string) (lastErr error) {
	s.Lock.Lock()
	defer s.Lock.Unlock()
	for _, uid := range uids {
		if err := websocket.JSON.Send(s.Conns[uid], msg); err != nil {
			lastErr = err
		}
	}
	return
}
```